src/common/reducers/repository-input.js

```diff
@@ -19,39 +13,15 @@ export function repositoryInput(state = {
       return {
         ...state,
         inputValue: action.payload,
-        repository: parseRepository(action.payload),
+        repository: parseGitHubRepoUrl(action.payload),
```

Comment:
Whole repository object (owner, name, fullName, url) is kept here.

I wonder if it would make sense to move repository info into separate small reducer that would handle this (instead of nested object).
That would also make repository info more independent from repository input component actions and could be reused elsewhere.